src/databricks/labs/ucx/assessment/sequencing.py

```diff
@@ -0,0 +1,349 @@
+                    maybe_task_dependency = self._nodes.get(task_dependency_key)
+                    if maybe_task_dependency:
+                        self._outgoing_references[task_key].add(maybe_task_dependency)
+                    else:
+                        problem = DependencyProblem(
+                            'task-dependency-not-found', f"Could not find task: {task_dependency}"
+                        )
+                        problems.append(problem)
+        return MaybeMigrationNode(job_node, problems)
+
+    def _register_workflow_task(self, task: jobs.Task, parent: MigrationNode) -> MaybeMigrationNode:
+        """Register a workflow task.
+
+        TODO:
+            Handle following jobs.Task attributes:
+            - for_each_task
+            - libraries
+            - notebook_task
+            - pipeline_task
+            - python_wheel_task
+            - run_job_tasl
+            - spark_jar_task
+            - spark_python_taks
+            - spark_submit_task
+            - sql_task
+        """
+        problems: list[DependencyProblem] = []
+        task_id = f"{parent.object_id}/{task.task_key}"
+        task_node = self._nodes.get(("TASK", task_id), None)
+        if task_node:
+            return MaybeMigrationNode(task_node, problems)
+        task_node = MigrationNode(
+            node_id=next(self._counter),
+            object_type="TASK",
+            object_id=task_id,
+            object_name=task.task_key,
+            object_owner=parent.object_owner,  # No task owner so use parent job owner
+        )
+        self._nodes[task_node.key] = task_node
+        # `task.new_cluster` is not handled because it is part of the task and not a separate node
+        if task.existing_cluster_id:
+            maybe_cluster_node = self._register_cluster(task.existing_cluster_id)
+            problems.extend(maybe_cluster_node.problems)
+            if maybe_cluster_node.node:
+                self._outgoing_references[task_node.key].add(maybe_cluster_node.node)
+        if task.job_cluster_key:
+            job_cluster_node = self._nodes.get(("CLUSTER", f"{parent.object_id}/{task.job_cluster_key}"))
+            if job_cluster_node:
+                self._outgoing_references[task_node.key].add(job_cluster_node)
+            else:
+                problem = DependencyProblem('cluster-not-found', f"Could not find cluster: {task.job_cluster_key}")
+                problems.append(problem)
+        return MaybeMigrationNode(task_node, problems)
+
+    def _register_job_cluster(self, cluster: jobs.JobCluster, parent: MigrationNode) -> MaybeMigrationNode:
+        """Register a job cluster.
+
+        A job cluster is defined within a job and therefore is found when defined on the job by definition.
+        """
+        # Different jobs can have job clusters with the same key, therefore job id is prepended to ensure uniqueness
```

Comment:
Note the differentiation between job clusters and "regular" clusters by prefixing the job id, similarly to tasks. The object_type is "CLUSTER" for both